packages/core/src/index.ts

```diff
@@ -44,37 +40,60 @@ export const resample = (state: State): State => {
   });
 };
 
-const step = (state: State, numSteps: number): State => {
+const step = (
+  state: State,
+  until: () => boolean
```

Comment:
Here `until` is a positional parameter rather than named; in contrast, in `ad.ts` I have the user pass `until` inside of an object:

https://github.com/penrose/penrose/blob/b3bbd666e70128601751517916fb909f58e73e0b/packages/core/src/types/ad.ts#L248-L251

This way, they have to write the word `until`, so in the code it's hopefully more clear that the function returning `true` is what's supposed to stop the stepping.

Could we do a similar thing here?